rubicon_ml/client/base.py

```diff
@@ -9,19 +19,22 @@ class Base:
         The config, which injects the repository to use.
     """
 
-    def __init__(self, domain, config=None):
+    def __init__(self, domain: DOMAIN_TYPES, config: Optional[Config] = None):
```

Comment:
I have to look into why the `config` is optional - that doesn't seem right

Reply:
Yeah, the default value here is `None`, do we need to make it required?

Reply:
not in this PR, maybe in the future. I think its a remnant of the fact that `Rubicon` used to extend the `Base` and didn't take in a config since it generated it. I'll verify that's the case and if so make an issue to fix it